### PR TITLE
fix: apps and jobs permission

### PR DESCRIPTION
**Pull request by Kishan789dev**
Fixes https://github.com/devtron-labs/devtron/issues/4846
### Desription
fixed the issue related user permission ,user having permisssion for the view only was able to create app/job

## Checklist:

* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.

-->
```release-note

```

### CODE REVIEW

File: api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go [hunk 1]
```diff
@@ -600,11 +600,13 @@ func (handler *PipelineConfigRestHandlerImpl) TriggerCiPipeline(w http.ResponseW
...
+	if ciPipeline.App.AppType == helper.Job {
 		appRbacOk = handler.enforcer.Enforce(token, casbin.ResourceJobs, casbin.ActionTrigger, appObject) && handler.enforcer.Enforce(token, casbin.ResourceWorkflow, casbin.ActionTrigger, workflowObject) && handler.enforcer.Enforce(token, casbin.ResourceJobsEnv, casbin.ActionTrigger, triggerObject)
-	}
 
+	} else {
+		appRbacOk = handler.enforcer.Enforce(token, casbin.ResourceApplications, casbin.ActionTrigger, appObject)
+	}
 	if !appRbacOk {
 		handler.Logger.Debug(fmt.Errorf("unauthorized user"), "Unauthorized User", http.StatusForbidden)
 		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusForbidden)
```
Comment (kartik-579) [inline api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go:612]: only valid for devtron applications, should skip for jobs

Comment (kartik-579) [reply api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go:612]: from line ~610 to 633

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go (hunk 1 above)
Comment (Kishan789dev) [reply api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go:612]: changes incorporated

**Review by vikramdevtron — APPROVED** [review]

File: api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go (hunk 1 above)
Comment (Kishan789dev) [reply api/restHandler/app/pipeline/configure/BuildPipelineRestHandler.go:628]: changes done

**Review by kartik-579 — APPROVED** [review]

**Pull request merged**